### PR TITLE
Add tests for SchemeSpec and some missing Repl functionality

### PR DESCRIPTION
**Pull request by jmcardon**
PR checklist:
* [x] Test coverage for the proposed changes

### CODE REVIEW

File: pact-request-api/Pact/Core/Command/Client.hs [hunk 1]
```diff
@@ -22,11 +77,682 @@ import Pact.Core.Command.Types
...
+data ApiPublicMeta = ApiPublicMeta
+  { _apmChainId :: Maybe ChainId
+  , _apmSender :: Maybe Text
+  , _apmGasLimit :: Maybe GasLimit
+  , _apmGasPrice :: Maybe GasPrice
+  , _apmTTL :: Maybe TTLSeconds
+  , _apmCreationTime :: Maybe TxCreationTime
+  } deriving (Eq, Show, Generic)
+
+(.:@?) :: JD.FromJSON (StableEncoding b) => JD.Object -> JD.Key -> A.Parser (Maybe b)
```
Comment (rsoeldner) [inline pact-request-api/Pact/Core/Command/Client.hs:174]: 😭 ... ok

File: pact-request-api/Pact/Core/Command/Client.hs [hunk 2]
```diff
@@ -22,11 +77,682 @@ import Pact.Core.Command.Types
...
+instance J.Encode ApiPublicMeta where
+  build o = J.object
+    [ "creationTime" J..?= (StableEncoding <$> _apmCreationTime o)
+    , "ttl" J..?= (StableEncoding <$> _apmTTL o)
+    , "gasLimit" J..?= (StableEncoding <$> _apmGasLimit o)
+    , "chainId" J..?= (StableEncoding <$> _apmChainId o)
+    , "gasPrice" J..?= (StableEncoding <$> _apmGasPrice o)
+    , "sender" J..?= _apmSender o
+    ]
+  {-# INLINABLE build #-}
```
Comment (rsoeldner) [inline pact-request-api/Pact/Core/Command/Client.hs:198]: maybe we should add a general TODO to revisit all inlinable pragmas\add them

File: pact/Pact/Core/StableEncoding.hs [hunk 3]
```diff
@@ -524,24 +532,54 @@ instance JD.FromJSON (StableEncoding GasPrice) where
...
-    ParsedInteger creationTime <- o JD..: "creationTime"
-    pure $ StableEncoding $ PublicMeta (ChainId chainId) sender gasLimit gasPrice (TTLSeconds ttl) (TxCreationTime creationTime)
+    StableEncoding ttl <- o JD..: "ttl"
+    StableEncoding creationTime <- o JD..: "creationTime"
+    pure $ StableEncoding $ PublicMeta (ChainId chainId) sender gasLimit gasPrice ttl creationTime
+
+instance J.Encode (StableEncoding a) => J.Encode (StableEncoding (Maybe a)) where
+  build (StableEncoding a) = J.build (StableEncoding <$> a)
 
+-- instance J.Encode (StableEncoding a) => J.Encode (StableEncoding [a]) where
```
Comment (rsoeldner) [inline pact/Pact/Core/StableEncoding.hs:584]: leftover

File: repl/Main.hs [hunk 4]
```diff
@@ -8,35 +6,138 @@ import Pact.Core.Builtin
...
+-- Todo: trace output and coverage?
+loadFlag :: O.Parser ReplOpts
+loadFlag = fmap OLoad $
+  ReplLoadFile
+    <$> O.flag False True
+        (O.short 'r' <> O.long "findscript" <>
+        O.help "For .pact files, attempts to locate a .repl file to execute.")
+    <*> O.flag False True
+        (O.short 't' <> O.long "trace" <> O.help "Show trace output")
+    -- <*> O.flag False True
```
Comment (rsoeldner) [inline repl/Main.hs:69]: tracked here: https://app.asana.com/0/1204069832833999/1207760643974332/f

File: pact-request-api/Pact/Core/Command/Client.hs (hunk 1 above)
Comment (jmcardon) [reply pact-request-api/Pact/Core/Command/Client.hs:174]: hahaha we can remove this, It just saved me some lines of code.

**Review by rsoeldner — APPROVED** [review]

**Pull request merged**